src/electionguard-ui/ElectionGuard.UI/ViewModels/CreateMultiTallyViewModel.cs

```diff
@@ -0,0 +1,353 @@
+            var ballotAddedTotal = 0L;
+            var ballotChallengedTotal = 0L;
+            var ballotSpoiledTotal = 0L;
+            var ballotDuplicateTotal = 0L;
+            var ballotRejectedTotal = 0L;
+
+            allUploads.ForEach((upload) =>
+            {
+                ballotCountTotal += upload.BallotCount;
+                ballotAddedTotal += upload.BallotImported;
+                ballotChallengedTotal += upload.BallotChallenged;
+                ballotSpoiledTotal += upload.BallotSpoiled;
+                ballotDuplicateTotal += upload.BallotDuplicated;
+                ballotRejectedTotal += upload.BallotRejected;
+            });
+            var electionItem = new ElectionItem
+            {
+                Election = election,
+                TallyId = tallyId,
+                BallotUploads = new(allUploads),
+                BallotAddedTotal = ballotAddedTotal,
+                BallotChallengedTotal = ballotChallengedTotal,
+                BallotDuplicateTotal = ballotDuplicateTotal,
+                BallotRejectedTotal = ballotDuplicateTotal,
+                BallotSpoiledTotal = ballotSpoiledTotal,
+                BallotCountTotal = ballotCountTotal
+            };
+            Elections.Add(electionItem);
+        }
+    }
+
+    private async Task FillKeyCeremonies()
+    {
+        var allKeys = await _keyCeremonyService.GetAllCompleteAsync();
+        foreach (var item in allKeys)
+        {
+            var count = await _electionService.CountByKeyCeremonyIdAsync(item.KeyCeremonyId);
+            if(count > 1)
+            {
+                KeyCeremonies.Add(item);
+            }
+        }
+    }
+
+    [RelayCommand]
+    private void SelectionChanged()
+    {
+        if(SelectedElections.Count != Elections.Count)
+        {
+            SelectAll = false;
+        }
+        CreateTalliesCommand.NotifyCanExecuteChanged();
+        JoinTalliesCommand.NotifyCanExecuteChanged();
+    }
+
+    private async Task<string> CreateNewTally(ElectionItem election, bool multi=false)
+    {
+        ElectionGuardException.ThrowIfNull(election, $"Could not load election selected");
+        ElectionGuardException.ThrowIfNull(election.Election, $"ElectionItem does not contain an election");
+        ElectionGuardException.ThrowIfNull(election.Election.ElectionId, $"Election does not have an election id");
+        ElectionGuardException.ThrowIfNull(election.Election.KeyCeremonyId, $"Election does not have a key ceremony id");
+
+        // calculate ballot count and upload count
+        var ballotCount = await _ballotService.GetCountBallotsByElectionIdStateAsync(election.Election.ElectionId, BallotBoxState.Cast);
+        var challengedCount = await _ballotService.GetCountBallotsByElectionIdStateAsync(election.Election.ElectionId, BallotBoxState.Challenged);
+        var spoiledCount = await _ballotService.GetCountBallotsByElectionIdStateAsync(election.Election.ElectionId, BallotBoxState.Spoiled);

```

Comment:
There's no way to do this in a single call to mongo?

Reply:
best we can do for  now